doc/iterations/2025.4/encode_workbench_state_in_url.adoc

```diff
@@ -0,0 +1,109 @@
+= (M) Encode the workbench state in the URL
+
+Issue: https://github.com/eclipse-sirius/sirius-web/issues/4886[#4886]
+
+== Problem
+
+When opening a project, the URL always starts with: `/projects/:projectId/edit/`.
+
+If at least 1 representation editor is opened, then the URL starts with: `/projects/:projectId/edit/:representationId` where `representationId` is the ID of the representation opened in the active editor.
+
+And since https://github.com/eclipse-sirius/sirius-web/issues/4376[#4376], the URL also contains the current selection of the workbench, meaning the URL can be of the form `/projects/:projectId/edit/?selection=id1%2Cid2` or `/projects/:projectId/edit/:representationId?selection=id1%2Cid2` (where `%2C` encodes `,`).
+
+Reversely, when resolving one of the above-mentioned URLs, the corresponding project, representation and selections are set.
+
+However, there are parts of the workbench state which are not encoded in the URL.
+
+These parts are the views opened in the left and right panels (e.g. in Sirius Web application, between `Explorer` and `Validation` on the left, and between `Details`, `Query`, `Representations` and `Related Elements` on the right).
```

Comment:
Just a detail but the views available can be changed in downstream applications which may remove default views or even add new ones with different configurations. A downstream application can even change the position of a view (explorer on the right for example)